packages/webapp/src/_app/eos/api.ts

```diff
@@ -30,6 +30,7 @@ export const getRow = async <T>(
     options?: TableQueryOptions
 ): Promise<T | undefined> => {
     options = { ...TABLE_PARAM_DEFAULTS, ...options };
+    options.lowerBound = options.keyValue;
```

Comment:
Another option: `options = { ...TABLE_PARAM_DEFAULTS, ...options, lowerBound: options?.keyValue };`
Just a comment...either way technically works.

That said, something feels weird about replacing `lowerBound` here with `keyValue`. For example, any time `getTableRows` is called via `getRow`, `lowerBound` (if passed in) will be discarded.

I didn't write this portion of the code in the first place; it'd be good for us to wait for @sparkplug0025 to weigh in before merging. Otherwise, I can do a deeper dive and give it some more thought tomorrow.

I think that before, `getRow` and `getTableRows` had distinct APIs (param types). Now, with the introduction of the `TableQueryOptions` interface, they're converging. And it seems we have two keys (`lowerBound` and `keyValue`) that basically do the same thing. We should probably try to avoid that.

Reply:
Well, lowerBound is just that, and what we're saying is that we want the first record that matches the value. I'll certainly wait for @sparkplug0025 to weigh in, but this does emulate what was there before and I think just represents an odd use of bounds. This query *should* only ever return one record, but I'm assuming that by specifying the sought value as the lowerBound, we're ensuring that the first item with this value gets returned.

Reply:
As to the assignment being part of the default value application, I felt they were 2 different operations that were easier to understand when kept separate, especially given the esoteric oddity of the 2nd line. I didn't want that all muddled in the same line.

Reply:
`keyValue` and `keyName` are only consumed (as convenience properties) by `getRow(...)`. They're not actually consumed by `getTableRows(...)` directly...that's what's throwing me off. What if we separated it out in the interface. So `TableQueryOptions` would just be:

```
export interface TableQueryOptions {
    scope?: string;
    lowerBound?: any;
    upperBound?: any;
    limit?: number;
}
```

and the type on `getRow` for `options`:

```
export const getRow = async <T>(
    table: string,
    options?: { keyName?: string; keyValue?: string; } & TableQueryOptions,
): Promise<T | undefined> => {
```

That way, it's clear that `keyName` and `keyValue` are only for `getRow()` and not `getTableRows()`?

Reply:
The way that I see this is we have two functions:
1. `getRow`: to get a single record based on the key (like a pretty simple get from a kv store)
2. `getRows`: to effectively setup, execute a query and work on the results.

That being said I'd probably remove the `TableQueryOptions` altogether from `getRow` because it just does not make sense to set these query options in a simple get. Then I'd rollback and keep using the original interface `getRow(table, keyName, keyValue)`.

Also, on top of that we could rename `getRows` to `queryEosTable` so we don't make any confusion between them.

`getRow` needs to fetch the first record from the `lowerbound` results (returned by `getRows`) and see if it matches with the expected key (hence the limit 1 and keyName comparison).